### PR TITLE
UF-8395: Prepare for shifting namespace shortcode from code to DB

### PR DESCRIPTION
**Pull request by ronnberg**

### CODE REVIEW

File: src/test/java/se/sundsvall/supportmanagement/service/config/NamespaceConfigServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+class NamespaceConfigServiceTest {
+
+	@Mock
+	private NamespaceConfigRepository configRepositoryMock;
+
+	@Mock
+	private NamespaceConfigMapper mapperMock;
+
+	@Captor
+	ArgumentCaptor<NamespaceConfigEntity> entityCaptor;
```
Comment (foobar-1337) [inline src/test/java/se/sundsvall/supportmanagement/service/config/NamespaceConfigServiceTest.java:39]: private?

File: src/test/java/se/sundsvall/supportmanagement/integration/db/model/NamespaceConfigEntityTest.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.core.AllOf.allOf;
+
+
+class NamespaceConfigEntityTest {
+
+	@BeforeAll
+	static void setup() {
+		final var random = new Random();
```
Comment (foobar-1337) [inline src/test/java/se/sundsvall/supportmanagement/integration/db/model/NamespaceConfigEntityTest.java:27]: No need to declare a variable here?

`registerValueGenerator(() -> now().plusDays(new Random().nextInt()), OffsetDateTime.class);`

File: pom.xml [hunk 3]
```diff
@@ -121,7 +121,7 @@
 				<groupId>org.apache.maven.plugins</groupId>
 				<artifactId>maven-surefire-plugin</artifactId>
 				<configuration>
-					<argLine>${surefireArgLine} -Xmx512m</argLine>
+					<argLine>${surefireArgLine} -Xmx1024m</argLine>
```
Comment (foobar-1337) [inline pom.xml:124]: Do you need to increase this when `spring.test.context.cache.maxSize` is set (decreased)?

File: src/integration-test/resources/NamespaceConfigIT/__files/test04_updateConfig/response_after.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+  "shortCode" : "NS1-UPDATE",
+  "created": "2021-12-31T23:59:59.999+01:00",
+  "modified": "${json-unit.ignore}"
```
Comment (foobar-1337) [inline src/integration-test/resources/NamespaceConfigIT/__files/test04_updateConfig/response_after.json:4]: anyString instead of ignore?

**Review by cheezi0747 — APPROVED** [review]

File: src/test/resources/spring.properties [hunk 5]
```diff
@@ -0,0 +1 @@
+spring.test.context.cache.maxSize=10
```
Comment (foobar-1337) [inline src/test/resources/spring.properties:1]: This should be added in dept44 i.m.h.o.

File: src/integration-test/java/se/sundsvall/supportmanagement/apptest/NamespaceConfigIT.java [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+import static org.springframework.http.HttpStatus.NO_CONTENT;
+import static org.springframework.http.HttpStatus.OK;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+@WireMockAppTestSuite(files = "classpath:/NamespaceConfigIT/", classes = Application.class)
+@Sql({
+	"/db/scripts/truncate.sql",
+	"/db/scripts/testdata-it.sql"
+})
+public class NamespaceConfigIT extends AbstractAppTest {
```
Comment (foobar-1337) [inline src/integration-test/java/se/sundsvall/supportmanagement/apptest/NamespaceConfigIT.java:31]: Remove `public` ?

File: src/main/java/se/sundsvall/supportmanagement/api/NamespaceConfigResource.java [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+	public ResponseEntity<Void> updateNamespaceConfig(
+		@Parameter(name = "namespace", description = "Namespace", example = "my.namespace") @Pattern(regexp = NAMESPACE_REGEXP, message = NAMESPACE_VALIDATION_MESSAGE) @PathVariable final String namespace,
+		@Parameter(name = "municipalityId", description = "Municipality id", example = "2281") @ValidMunicipalityId @PathVariable final String municipalityId,
+		@Valid @NotNull @RequestBody NamespaceConfig namespaceConfig) {
+
+		service.replace(namespaceConfig, namespace, municipalityId);
+		return noContent().build();
+	}
+
+	@DeleteMapping()
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/supportmanagement/api/NamespaceConfigResource.java:98]: Remove `()` ?

File: pom.xml [hunk 8]
```diff
@@ -12,7 +12,7 @@
 	</parent>
 	<groupId>se.sundsvall</groupId>
 	<artifactId>api-service-support-management</artifactId>
-	<version>5.8</version>
+	<version>5.9</version>
 	<name>api-service-support-management</name>
 	<properties>
 		<app.jvm.xmx.mb>512</app.jvm.xmx.mb>
```
Comment (foobar-1337) [inline pom.xml:18]: Can be removed

File: pom.xml (hunk 3 above)
Comment (AndreSoderlund) [reply pom.xml:124]: This is set in gitlab, in the values.yaml. So I don't think this is needed at all, except if you are running it locally i guess.

Comment (foobar-1337) [reply pom.xml:124]: This is the sureFireArgLine (settings used during surefire-execution).
The value in values.yaml corresponds to "app.jvm.xmx.mb" in this pom.xml (can also be removed).

File: src/test/resources/spring.properties (hunk 5 above)
Comment (ronnberg) [reply src/test/resources/spring.properties:1]: do it

**Review by AndreSoderlund — APPROVED** [review]

**Pull request merged**